File: game_string.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++

class Solution{
public:
    int minValue(string s, int k){
        // code here
        int count[26] = {0};
        for(int i = 0; i < s.size(); ++i) {
            ++count[s[i]-'a'];
        }
        multiset<int> ms;
        for(int i = 0; i < 26; ++i) {
            if (count[i] != 0) {
                ms.insert(count[i]);
            }
        }
        for(int i=0; i<k; ++i) {
            multiset<int>::iterator it = ms.end();
            --it;
            int temp = *it;
            ms.erase(it);
            ms.insert(temp-1);
        }
        int ans = 0;
        for(auto item : ms) {
            ans += item * item;
        }
        return ans;
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        string s;
        int k;
        cin>>s>>k;
        
        Solution ob;
        cout<<ob.minValue(s, k)<<"\n";
    }
    return 0;
}
// } Driver Code Ends

File: string/game_string.py
#User function Template for python3
from collections import Counter
import heapq

class Solution:
    def minValue(self, s, k):
        # code here
        c = Counter(s)
        pq = []
        for _, v in c.items():
            heapq.heappush(pq, -v)

        for _ in range(k):
            heapq.heapreplace(pq, pq[0]+1)
            
        return sum(a * a for a in pq)

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        s = input()
        k = int(input())
        
        ob = Solution()
        print(ob.minValue(s, k))
# } Driver Code Ends
